server/MusicCrawler.Plex/Services/Singletons/PlexRepo.cs

```diff
@@ -24,7 +24,20 @@ public Task<ArtistPackage[]> QueryAllData()
     public async Task<ArtistMetadata[]> QueryAllArtistMetadata()
     {
         var plexLibraries = await _plexApi.GetLibraries();
-        var plexLibrary = plexLibraries.Where(it => it.Type == "artist").TakeRandomly(1).First(); // TODO: Probably should select a predefined library.
+        PlexLibrary plexLibrary;
+        try
+        {
+            plexLibrary =
+                plexLibraries
```

Comment:
`First()` already throws an exception if one isn't found (You'd want to call `FirstOrDefault` if you wanted null to bubble out for the Let to check.

But think a snippet like this would be preferable?
```cs
var env = Environment.GetEnvironmentVariable("preferredPlexLibrary");

// ....

if (env == null)
{
    plexLibrary = plexLibraries.Where(it => it.Type == "artist").TakeRandomly(1).First();
}
else
{
   plexLibrary = plexLibraries.First(it => string.Equals(it.Title, env));
   Console.WriteLine("Successfully found preferred plexLibrary:" + plexLibrary.Title);
}
```
Where the env variable can probably be retrieved on some init step (maybe the ctor?)

i like the explicit null check on the env variable as it makes it clearer to me that we're just checking if the preference is specified or not.  The throwing logic does the same job, but requires more try/catches and makes the logic a bit more opaque and feels like there's errors happening when it's really just that we're seeing if the user set a preference or not

Reply:
I wrote out a commit with the requested changes, but it got a bit more complex because there are more points of failure to handle, and it ended up at a state that I didn't really like.

My main issues are:
1-You do not get a stacktrace when the warning occurs, so it's less debuggable.
2-There are more variables and logic branches to keep track of, so it's less maintainable.

Do you mind if I revert the commit?